File: Inventory-Management-System/src/main/java/com/brocode/entity/Supplier.java
package com.brocode.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.util.List;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
@EqualsAndHashCode(callSuper = true)
public class Supplier extends BaseEntity{
    private String name;
    private String contact;
    private String address;
    private String contactPerson;

    @OneToMany(mappedBy = "supplier", cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    private List<Order> orders;
}

File: mobile-application/app/src/main/java/com/example/inventory/data/Supplier.kt
package com.example.inventory.data

data class SupplierResponseDto(
    val name: String,
    val contact: String,
    val contactName: String,
    val address: String,
    val orders: List<OrderResponseDto>
): BaseResponseDto()

data class SupplierCreateDto(
    val name: String,
    val contact: String,
    val contactName: String,
    val address: String
)